primo/utils/map_utils.py

```diff
@@ -22,548 +21,352 @@
-    selected: pd.DataFrame, map_obj: folium.Map, cluster_colors: Dict[str, str]
-) -> None:
-    """
-    Add project markers to a folium map.
-
-    Parameters
-    ----------
-    selected : pd.DataFrame
-        DataFrame containing selected project data
-    map_obj : folium.Map
-        Folium map object
-    cluster_colors : Dict[str, str]
-        Dictionary mapping projects to colors
-
-    Returns
-    -------
-    None
-    """
-    gas_index = selected.columns.get_loc("Gas [Mcf/Year]") + 1
-    oil_index = selected.columns.get_loc("Oil [bbl/Year]") + 1
-    score_index = selected.columns.get_loc("Priority Score [0-100]") + 1
-    for row in selected.itertuples():
-        if pd.isna(row.Project):
-            continue
-        gas = row[gas_index]
-        oil = row[oil_index]
-        cluster = "Project: " + str(row.Project)
-        score = row[score_index]
-        popup_text = (
-            f"Oil [bbl/day]: {oil}<br>Gas [Mcf/day]: {gas}<br>"
-            f"Candidate Project: {cluster}<br>Score: {score}<br>"
-        )
-        color = cluster_colors.get(
-            row.Project, "gray"
-        )  # Use 'gray' if cluster not in color scheme
-        folium.CircleMarker(
-            location=[row.Latitude, row.Longitude],
-            radius=5,
-            popup=popup_text,
-            fill=True,
-            color=color,
-        ).add_to(map_obj)
-
-
-def visualize_data_with_projects(
-    selected: pd.DataFrame, state_shapefile: gpd.GeoDataFrame
-) -> folium.Map:
-    """
-    Visualize data with projects by adding project markers to a folium map.
-
-    Parameters
-    ----------
-    selected : pd.DataFrame
-        DataFrame containing selected project data
-    state_shapefile : gpd.GeoDataFrame
-        GeoDataFrame containing shapefile data
-
-    Returns
-    -------
-    folium.Map
-        Folium map object
-    """
-    map_obj = folium.Map(
-        location=get_mean_centroid(state_shapefile),
-        zoom_start=7,
-    )
-    common_visualization(map_obj, state_shapefile)
-    cluster_list = pd.unique(selected["Project"])
-    num_cluster = len(cluster_list)
-    cluster_colors = get_cluster_colors(num_cluster, cluster_list)
-    add_project_markers_to_map(selected, map_obj, cluster_colors)
-    return map_obj
+            if county_name is None:
+                raise AttributeError("None of the county name attributes are found.")
+
+            centroid = [county.geometry.centroid.y, county.geometry.centroid.x]
+            folium.map.Marker(
+                location=centroid,
+                icon=folium.DivIcon(
+                    html=(
+                        f'<div style="font-size: 11pt; color: black; text-align: center; '
+                        f'font-weight: bold;">{county_name}</div>'
+                    )
+                ),
+            ).add_to(map_obj)
+
+        # Create legend
+        if legend is True:
+            gas_legend = '<i style="color:red">o - Gas Well</i>'
```

Comment:
Is it possible we can make the "o - Gas Well" section an argument to the function? For example,  gas_legend = '<i style="color:red">' + legend_string + '</i>', where legend_string is an argument and 'o - Gas Well' is the default value